scripts/extensions/annotationsLibrary/src/extension.tsx

```diff
@@ -0,0 +1,48 @@
+import {ISuperdesk, IExtension, IRestApiResponse} from 'superdesk-api';
+import {getAnnotationInputWithKnowledgeBase} from './AnnotationInputWithKnowledgeBase';
+import {getAnnotationsLibraryPage} from './annotations-library-page';
+import {getFields} from './GetFields';
+import {IKnowledgeBaseItem} from './interfaces';
+
+function annotationFromLibraryTabSelectedByDefault(superdesk: ISuperdesk, annotationText: string) {
+    const {dataApi} = superdesk;
+    const {nameField} = getFields(superdesk);
+    const {generateFilterForServer} = superdesk.forms;
+
+    return dataApi.query<IKnowledgeBaseItem>(
+        'concept_items',
+        1,
+        {field: 'name', direction: 'ascending'},
+        {name: generateFilterForServer(nameField.type, annotationText)},
+    ).then((res: IRestApiResponse<IKnowledgeBaseItem>) => res._items.length > 0);
+}
+
+var extension: IExtension = {
+    activate: (superdesk: ISuperdesk) => {
+        const {gettext} = superdesk.localization;
+
+        return Promise.resolve({
```

Comment:
You don't need `Promise.resolve` if you declare the function as `async`, I know that we don't use `async/await` but at least `async` is just very convenient and readable, as you immediately know if it will return a Promise

Reply:
I don't mind the resolve. Async is a new thing and more developers understand promises than async/await. I wouldn't mind switching if there were significant benefits, but I don't see them yet. You'll have to up your selling skills :smile:

Reply:
I actually learned first async functions than Promise.resolve so maybe I'm biased 😁 But still I think it's way more readable, returning a promise is only a way of saying the function is async. Also, with async you make sure you always have a promise as a result. If the function has several `return`, one of them might be a promise, other one can be a value...

Reply:
> Also, with async you make sure you always have a promise as a result.

TypeScript makes sure of this for me. If you declare that function returns a promise, it won't compile if you try to return a value.

> I think it's way more readable

I'd argue that for most developers it's less readable. For 20 years JavaScript has been around if you saw `return {a: 1}` you could be 99% sure that the function was synchronous. 1% goes to people who don't know what they are doing and write Schrodinger functions which are used in both ways at the same time.

Still not buying it :smile:

Reply:
Yep typescript mitigates that issue, I was talking about Javascript in general. Anyway use whatever you want 😄 I just think dealing with promises is kind of like doing the work `async/await` abstract you from doing, even tho know it is doing the same thing. Kind of like writing a `class` instead of a `function`

Reply:
It's not about what _I want_, but rather what enables us to write better code. I think we agree that in this case, the main difference is in syntax. I assume people will be more familiar with promises since that's what we currently use and promises have been around for longer. If you convince others I don't mind starting using async/await in the project :slightly_smiling_face: